testing/fuchsia/run_tests.py

```diff
@@ -102,18 +104,31 @@ def bundled_test_runner_of(target_id: str) -> BundledTestRunner:
           tests
       )
   )
+  packages = []
   for test in tests:
-    original_package = test['package']
-    test['package'] = os.path.join(
-        OUT_DIR, test['package'].replace('-0.far', '.far')
-    )
-    try:
-      os.remove(test['package'])
-    except FileNotFoundError:
-      pass
-    os.symlink(original_package, test['package'])
+    if 'package' in test:
+      packages.append(test['package'])
+    else:
+      assert 'packages' in test, \
+             'Expect either one package or a list of packages'
+      packages.extend(test['packages'])
+  resolved_packages = []
+  for package in list(dict.fromkeys(packages)):
+    if package.endswith('-0.far'):
+      # Make a symbolic link to match the name of the package itself without the
+      # '-0.far' suffix.
+      original_package = package
+      package = os.path.join(OUT_DIR, package.replace('-0.far', '.far'))
+      try:
+        os.remove(package)
+      except FileNotFoundError:
+        pass
+      os.symlink(original_package, package)
+      resolved_packages.append(package)
+    else:
+      resolved_packages.append(os.path.join(OUT_DIR, package))
   return BundledTestRunner(
-      target_id, [test['package'] for test in tests],
+      target_id, list(dict.fromkeys(resolved_packages)),
```

Comment:
Why is it necessary here and elsewhere to have the collection of packages be a list -> dict -> list again?

If we're worried about duplicates, it should be a set; if duplicates aren't possible, then we should just keep it a list.